rosidl_generator_c/resource/idl__description.c.em

```diff
@@ -0,0 +1,140 @@
+  full_type_descriptions.append(extract_subinterface(send_goal_service, 'response_message'))
+  full_type_descriptions.append(extract_subinterface(send_goal_service, 'event_message'))
+
+  get_result_service = extract_subinterface(type_description_msg, 'get_result_service')
+  full_type_descriptions.append(get_result_service)
+  full_type_descriptions.append(extract_subinterface(get_result_service, 'request_message'))
+  full_type_descriptions.append(extract_subinterface(get_result_service, 'response_message'))
+  full_type_descriptions.append(extract_subinterface(get_result_service, 'event_message'))
+
+  full_type_descriptions.append(extract_subinterface(type_description_msg, 'feedback_message'))
+
+def typename_to_c(typename):
+  return typename.replace('/', '__')
+
+def static_seq(varname, values):
+  """Statically define a runtime Sequence or String type."""
+  if values:
+    return f'{{{varname}, {len(values)}, {len(values)}}}'
+  return '{NULL, 0, 0}'
+}@
+
+#include "@(include_base)__struct.h"
+
+// Declare and define all type names, field names, and default values
+@[for itype_description in all_type_descriptions]@
+@{
+td_c_typename = typename_to_c(itype_description['type_name'])
+}@
+static char @(td_c_typename)__TYPE_NAME[] = "@(itype_description['type_name'])";
+@[  for field in itype_description['fields']]@
+static char @(td_c_typename)__FIELD_NAME__@(field['name'])[] = "@(field['name'])";
+@[    if field['default_value']]@
+static char @(td_c_typename)__DEFAULT_VALUE__@(field['name'])[] = "@(escape_string(field['default_value']))";
+@[    end if]@
+@[  end for]@
+
+@[end for]@
+@
+/// Define all arrays of Fields
+@[for itype_description in all_type_descriptions]@
+@{
+td_c_typename = typename_to_c(itype_description['type_name'])
+}@
+
+@[  if itype_description['fields']]@
+static rosidl_runtime_c__type_description__Field @(td_c_typename)__FIELDS[] = {
+@[    for field in itype_description['fields']]@
+  {
+    @(static_seq(f"{td_c_typename}__FIELD_NAME__{field['name']}", field['name'])),
+    {
+      @(field['type']['type_id']),  // TODO(ek) should this be named to FieldType const, or is value ok?
+      @(field['type']['capacity']),
+      @(field['type']['string_capacity']),
+      @(static_seq(f"{typename_to_c(field['type']['nested_type_name'])}__TYPE_NAME", field['type']['nested_type_name'])),
+    },
+    @(static_seq(f"{td_c_typename}__DEFAULT_VALUE__{field['name']}", field['default_value'])),
+  },
+@[    end for]@
+};
+@[  end if]@
+@[end for]@
+
+/// Define all IndividualTypeDescriptions
+@[for itype_description in all_type_descriptions]@
+@{
+td_c_typename = typename_to_c(itype_description['type_name'])
+}@
+
+static const rosidl_runtime_c__type_description__IndividualTypeDescription @(td_c_typename)__INDIVIDUAL_TYPE_DESCRIPTION = {
+  @(static_seq(f'{td_c_typename}__TYPE_NAME', itype_description['type_name'])),
+  @(static_seq(f'{td_c_typename}__FIELDS', itype_description['fields'])),
+};
```

Comment:
Same question/comment here; redefining these everywhere is going to bloat the binary sizes.